From d9200d1018ac916b30433da23898c8c5fbde0f28 Mon Sep 17 00:00:00 2001
From: Jordan Borean <jborean93@gmail.com>
Date: Tue, 18 Jul 2023 08:57:11 +1000
Subject: [PATCH] Fix Cython 3 Compatibility

Fixes compatibility with Cython 3 by fixing up the import so it doesn't
try and do a recursive import. This also increases the upper bound for
Cython to 4.0.0 now that 3.0.0 is compatible.

Signed-off-by: Jordan Borean <jborean93@gmail.com>
---
 gssapi/raw/_enum_extensions/ext_dce.pyx | 2 +-
 pyproject.toml                          | 2 +-
 setup.py                                | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/gssapi/raw/_enum_extensions/ext_dce.pyx b/gssapi/raw/_enum_extensions/ext_dce.pyx
index fd2989ae..190c8285 100644
--- a/gssapi/raw/_enum_extensions/ext_dce.pyx
+++ b/gssapi/raw/_enum_extensions/ext_dce.pyx
@@ -1,6 +1,6 @@
 from gssapi.raw.cython_types cimport OM_uint32
 
-import gssapi.raw._enum_extensions as ext_registry
+from gssapi.raw import _enum_extensions as ext_registry
 
 
 cdef extern from "python_gssapi_ext.h":
diff --git a/pyproject.toml b/pyproject.toml
index 2f5c208c..5ee26df6 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [build-system]
 requires = [
-    "Cython >= 0.29.29, < 3.0.0",  # 0.29.29 includes fixes for Python 3.11
+    "Cython >= 0.29.29, < 4.0.0",  # 0.29.29 includes fixes for Python 3.11
     "setuptools >= 40.6.0",  # Start of PEP 517 support for setuptools
 ]
 build-backend = "setuptools.build_meta"
diff --git a/setup.py b/setup.py
index 36c7aafa..881b9f9f 100755
--- a/setup.py
+++ b/setup.py
@@ -274,7 +274,7 @@ def gssapi_modules(lst):
 
 setup(
     name='gssapi',
-    version='1.8.2',
+    version='1.8.3',
     author='The Python GSSAPI Team',
     author_email='jborean93@gmail.com',
     packages=['gssapi', 'gssapi.raw', 'gssapi.raw._enum_extensions',
